ddl/label/rule.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package label
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/util/codec"
+	"gopkg.in/yaml.v2"
+)
+
+// IDPrefix is the prefix for label rule ID.
+const IDPrefix = "schema"
+
+var (
+	// TableIDFormat is the format of the label rule ID for a table.
+	// The format follows "schema/database_name/table_name".
+	TableIDFormat = "%s/%s/%s"
+)
+
+// Rule is used to establish the relationship between labels and a key range.
+type Rule struct {
+	ID       string      `json:"id"`
+	Labels   Labels      `json:"labels"`
+	RuleType string      `json:"rule_type"`
+	Rule     interface{} `json:"rule"`
+}
+
+// ApplyAttributesSpec will transfer attributes defined in AttributesSpec to the labels.
+func (r *Rule) ApplyAttributesSpec(spec *ast.AttributesSpec) error {
+	// construct a string list
+	attrBytes := []byte("[" + spec.Attributes + "]")
```

Comment:
What if `spec.Attributes[0] == '['`?

Reply:
`alter table x attributes="["` will be regarded as an invalid attribute.

Reply:
Is there an white list or black list for attributes? What is the definition of valid?

Reply:
No, but it needs to meet the requirement of yaml.